client/src/debug/FabricDebugConfigurationProvider.ts

```diff
@@ -81,7 +81,12 @@ export class FabricDebugConfigurationProvider implements vscode.DebugConfigurati
                 config.program = path.join(folder.uri.fsPath, 'node_modules', '.bin', 'fabric-chaincode-node');
             }
 
-            const tsFiles: Array<vscode.Uri> = await vscode.workspace.findFiles(new vscode.RelativePattern(folder, '**/*.ts'), '**/node_modules/**', 1);
+            const allTsFiles: Array<vscode.Uri> = await vscode.workspace.findFiles(new vscode.RelativePattern(folder, '**/*.ts'), '**/node_modules/**', 1);
+
+            // Ignore any TypeScript test files in a JavaScript project
+            const tsFiles: Array<vscode.Uri> = allTsFiles.filter((file: vscode.Uri) => {
+                return !file.fsPath.endsWith('.test.ts');
```

Comment:
Isn't this a naming convention rather than actually being required?

Reply:
It would've picked up on our generated tests, but to be safe we'll now search for a `tsconfig.json` before assuming a project is TypeScript.